app/uk/gov/hmrc/tai/connectors/CacheConnector.scala

```diff
@@ -16,36 +16,22 @@
 
 package uk.gov.hmrc.tai.connectors
 
+import cats.data.OptionT
+import cats.implicits._
 import com.google.inject.{Inject, Singleton}
-import play.Logger
 import play.api.Configuration
-import play.api.libs.json.{JsResultException, JsValue, Json, Reads, Writes}
-import play.modules.reactivemongo.ReactiveMongoComponent
-import uk.gov.hmrc.cache.repository.CacheMongoRepository
+import play.api.libs.json._
 import uk.gov.hmrc.crypto.json.{JsonDecryptor, JsonEncryptor}
 import uk.gov.hmrc.crypto.{ApplicationCrypto, CompositeSymmetricCrypto, Protected}
 import uk.gov.hmrc.tai.config.MongoConfig
 import uk.gov.hmrc.tai.model.nps2.MongoFormatter
-import cats.implicits._
-import cats.data.OptionT
-import uk.gov.hmrc.cache.model.Cache
 
 import scala.concurrent.{ExecutionContext, Future}
 
-@Singleton
-class TaiCacheRepository @Inject()(mongo: ReactiveMongoComponent, mongoConfig: MongoConfig)(
-  implicit ec: ExecutionContext)
-  extends CacheMongoRepository("TAI", mongoConfig.mongoTTL)(mongo.mongoConnector.db, ec)
-
-class TaiCacheRepositoryUpdateIncome @Inject()(mongo: ReactiveMongoComponent, mongoConfig: MongoConfig)(
-  implicit ec: ExecutionContext)
-  extends CacheMongoRepository("TaiUpdateIncome", mongoConfig.mongoTTLUpdateIncome)(mongo.mongoConnector.db, ec)
-
-
 @Singleton
 class CacheConnector @Inject()(
-  cacheRepository: TaiCacheRepository,
-  cacheRepositoryUpdateIncome: TaiCacheRepositoryUpdateIncome,
+  taiCacheRepository: TaiCacheRepository,
+  taiUpdateIncomeCacheRepository: TaiUpdateIncomeCacheRepository,
   mongoConfig: MongoConfig,
```

Comment:
I already commented on this, but why is it a concern for CacheConnector to know about mongo config? 

I would push all the serialisation to the mongo layer rather than having it in the cache connector.

Reply:
👍 Added it to refactoring 

https://jira.tools.tax.service.gov.uk/browse/PAYEE-944